From isely@isely.net Sun Feb 10 18:23:34 2008
From: Mike Isely <isely@pobox.com>
Date: Sun, 10 Feb 2008 20:23:28 -0600 (CST)
Subject: USB: cypress_m8: Get rid of pointless NULL check
To: Lonnie Mendez <dignome@gmail.com>
Cc: Mike Isely at pobox <isely@pobox.com>, Greg KH <greg@kroah.com>, Marr <marr@copper.net>, linux-usb@vger.kernel.org
Message-ID: <Pine.LNX.4.64.0802102014230.3820@cnc.isely.net>

From: Mike Isely <isely@pobox.com>

Remove a NULL check in cypress_m8; the check is useless in this
context because it is referenced earlier in the same code path thus
the kernel would be oops'ed before reaching this point anyway.  (And
it's really pointless here anyway; if this pointer somehow is NULL the
driver is going to have serious problems in many other places.)

Signed-off-by: Mike Isely <isely@pobox.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/cypress_m8.c |   12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

--- a/drivers/usb/serial/cypress_m8.c
+++ b/drivers/usb/serial/cypress_m8.c
@@ -1397,13 +1397,11 @@ static void cypress_read_int_callback(st
 
 	spin_lock_irqsave(&priv->lock, flags);
 	/* check to see if status has changed */
-	if (priv != NULL) {
-		if (priv->current_status != priv->prev_status) {
-			priv->diff_status |= priv->current_status ^
-				priv->prev_status;
-			wake_up_interruptible(&priv->delta_msr_wait);
-			priv->prev_status = priv->current_status;
-		}
+	if (priv->current_status != priv->prev_status) {
+		priv->diff_status |= priv->current_status ^
+			priv->prev_status;
+		wake_up_interruptible(&priv->delta_msr_wait);
+		priv->prev_status = priv->current_status;
 	}
 	spin_unlock_irqrestore(&priv->lock, flags);
 
